assets/src/stories-editor/blocks/amp-story-page/copy-paste-handler.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { pasteHandler, serialize } from '@wordpress/blocks';
+import { documentHasSelection } from '@wordpress/dom';
+import { withDispatch, useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ensureAllowedBlocksOnPaste } from '../../helpers';
+
+function CopyPasteHandler( { children, onCopy, clientId, isSelected } ) {
+	const {
+		isFirstPage,
+		canUserUseUnfilteredHTML,
+	} = useSelect(
+		( select ) => {
+			const {
+				getBlockOrder,
+				getSettings,
+			} = select( 'core/block-editor' );
+			const { __experimentalCanUserUseUnfilteredHTML } = getSettings();
+			return {
+				isFirstPage: getBlockOrder().indexOf( clientId ) === 0,
+				canUserUseUnfilteredHTML: __experimentalCanUserUseUnfilteredHTML,
+			};
+		}, [ clientId ]
+	);
+
+	const { insertBlocks } = useDispatch( 'core/block-editor' );
+
+	const onPaste = ( event ) => {
+		// Ignore if the Page is not the selected page.
+		if ( ! isSelected ) {
+			return;
+		}
+		const clipboardData = event.clipboardData;
+
+		let plainText = '';
+		let html = '';
+
+		// IE11 only supports `Text` as an argument for `getData` and will
+		// otherwise throw an invalid argument error, so we try the standard
+		// arguments first, then fallback to `Text` if they fail.
+		try {
+			plainText = clipboardData.getData( 'text/plain' );
+			html = clipboardData.getData( 'text/html' );
+		} catch ( error1 ) {
+			try {
+				html = clipboardData.getData( 'Text' );
+			} catch ( error2 ) {
+				// Some browsers like UC Browser paste plain text by default and
```

Comment:
Should this error state, return a message to the user to explain that something went wrong?